runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/control/DefaultJobBundleFactory.java

```diff
@@ -255,6 +255,14 @@ public void close() throws Exception {
     // Clear the cache. This closes all active environments.
     // note this may cause open calls to be cancelled by the peer
     for (LoadingCache<Environment, WrappedSdkHarnessClient> environmentCache : environmentCaches) {
+      for (WrappedSdkHarnessClient client : environmentCache.asMap().values()) {
+        try {
+          client.close();
```

Comment:
Worth mentioning that this is added to close the environments irrespective of open bundles, since this will occur only during shutdown?

Reply:
I've removed this in favor of using `unref()`. I think I found the case where `unref()` would not be called: https://github.com/apache/beam/pull/10694/files#diff-e80c769f0011537cc2b60d3e7898cf5aR413